### PR TITLE
Add python code checks

### LINKED ISSUE #1493: Add code style checks to CI
Lets make our code more beautiful and lets add code style guidelines checks to Circle CI:

* Python: pep8 or flake8
* JavaScript: JSLint

### PR DESCRIPTION
**Pull request by xamanu**
Fixes #1493.

This PR adds [black](https://pypi.org/project/black/) and  [flake8](http://flake8.pycqa.org) code checks for backend code to the continous integration pipeline for the following file and directories:

* `manage.py`
* `server/`
* `tests/`
* `migrations/`

Further, this PR includes a code cleanup done with black and manual fixing of `flake8` spotted coding issues.

### CODE REVIEW

**Comment by xamanu** [comment]
Rebased after #1803

**Comment by xamanu** [comment]
Just added `black` checks to circle-ci, too. Hope you like it :smile:

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,13 @@
 alembic==0.9.2
 aniso8601==1.2.1
 bleach==2.0.0
+black==19.3b0
 cachetools==2.0.0
 certifi==2019.3.9
 chardet==3.0.4
 click==6.7
 coverage==4.3.4
+black==19.3b0
```
Comment (JorgeMartinezG) [inline requirements.txt]: The module is repeated twice

Comment (xamanu) [reply requirements.txt]: Oh, yes. Thanks. Fixed.

**Comment by xamanu** [comment]
> Also, we would need to inform the use of double quotes for strings in future PRs, right?
I included `black` in the code contribution documentation. Just by running it, as indicated, the single quotes are getting automatically transformed into `"`.

**Pull request merged**